### ISSUE TITLE
NGINX Ingress Controller - Error reading file /etc/nginx/GeoLiteCity.dat

### ISSUE THREAD

**Issue by amarruedo** [kubernetes/contrib#1451]
Hi!

I am experiencing the folowing issue with nginx ingress controller:

```
I0729 08:28:02.618643       1 event.go:216] Event(api.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"xxxx-ingress", UID:"33ca4b44-3df0-11e6-950a-42010af00123", APIVersion:"extensions", ResourceVersion:"400630", FieldPath:""}): type: 'Normal' reason: 'CREATE' ip: xxx.xxx.xxx.xxx
I0729 08:28:02.619053       1 event.go:216] Event(api.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"xxxx-ingress", UID:"33ca4b44-3df0-11e6-950a-42010af00123", APIVersion:"extensions", ResourceVersion:"400647", FieldPath:""}): type: 'Normal' reason: 'UPDATE' default/xxxx-ingress
E0729 08:28:02.988214       1 command.go:44] nginx error: signal: killed
E0729 08:28:02.990789       1 command.go:87] failed to execute nginx -s reload: Error reading file /etc/nginx/GeoLiteCity.dat
2016/07/29 08:28:02 [emerg] 19#19: GeoIP_open("/etc/nginx/GeoLiteCity.dat") failed in /etc/nginx/nginx.conf:29
nginx: [emerg] GeoIP_open("/etc/nginx/GeoLiteCity.dat") failed in /etc/nginx/nginx.conf:29
I0729 08:28:31.266472       1 main.go:193] Received SIGTERM, shutting down
```
Everything worked fine for me prior to version 0.7 of nginx controller (i.e. prior to introducing GeoIP). Now, it fails to reload nginx configuration and container crashes, which leads the nginx pod to a CrashLoopBackOff state.

I just changed the nginx image version in my deployment from 0.6 to 0.8.1 (I've tested with 0.7 as well, and I get the same error). 

Is there anything I'm missing?

Thanks in advance for your help.

**Comment by aledbf** [comment]
@amarruedo are you mounting a volume? The file exists in the image:
```
$ docker run gcr.io/google_containers/nginx-ingress-controller:0.8.2 ls /etc/nginx
GeoIP.dat
GeoLiteCity.dat
fastcgi.conf
fastcgi.conf.default
fastcgi_params
fastcgi_params.default
koi-utf
koi-win
lua
mime.types
mime.types.default
nginx.conf
nginx.conf.default
scgi_params
scgi_params.default
template
uwsgi_params
uwsgi_params.default
win-utf
```

**Comment by amarruedo** [comment]
@aledbf I'm not mounting any volume, and yes, I checked that the file exist inside the container. I thought that the problem could be related to a corrupt GeoLiteCity.dat, but if that was the case, then I woldn't be the only one experiencing this issue.

**Comment by amarruedo** [comment]
Well, I found the problem, reviewing for 100th time my nginx deployment specification I realized that I had  set memory limit of 10mb on the pod (classic copy/pasting from other deployment specification, my bad!), which I guess is not enough for GeoLiteCity.dat, and thus, it fails opening the database. 

I deleted memory realted resource limits from the delpoyment specification and now everything works as expected.

Thanks for the help.
